From cb51ec369186628e016a10ccf25b2b26d5d2ef85 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Mon, 10 Feb 2020 21:31:36 +0100
Subject: [PATCH] src/core/linux/SDL_threadprio.c: fix build without threads

Add an include on SDL_error.h to avoid the following build failure
without threads:

/home/buildroot/autobuild/instance-0/output-1/host/opt/ext-toolchain/bin/../lib/gcc/arm-buildroot-linux-uclibcgnueabihf/8.3.0/../../../../arm-buildroot-linux-uclibcgnueabihf/bin/ld: build/.libs/SDL_threadprio.o: in function `SDL_LinuxSetThreadPriority_REAL':
SDL_threadprio.c:(.text+0x0): undefined reference to `SDL_Unsupported'

Fixes:
 - http://autobuild.buildroot.org/results/7f7712c5bd47de4a3fcec1e0d0526fd5a3ecd532

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://bugzilla.libsdl.org/show_bug.cgi?id=4971]
---
 src/core/linux/SDL_threadprio.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/linux/SDL_threadprio.c b/src/core/linux/SDL_threadprio.c
index 162b1a3f1..2cbc4cb48 100644
--- a/src/core/linux/SDL_threadprio.c
+++ b/src/core/linux/SDL_threadprio.c
@@ -22,6 +22,7 @@
 
 #ifdef __LINUX__
 
+#include "SDL_error.h"
 #include "SDL_stdinc.h"
 
 #if !SDL_THREADS_DISABLED
-- 
2.24.1

